app/controllers/accounts_controller.rb

```diff
@@ -0,0 +1,49 @@
+class AccountsController < ApplicationController
+  def index
+    @accounts = current_user.accounts
+  end
+
+  def show
+    @account = Account.find(params[:account_number])
+  end
+
+  def new
+    account = Account.order(:account_number).last
+    !account ? next_number = 0 : next_number = account.account_number.to_i + 1
+    @account = current_user.accounts.build(account_number: next_number)
```

Comment:
why build an object?

Reply:
Without the object, there are errors coming up in the view of this function. I checked online and mostly the new function is done this way.

Reply:
what error?